### DISCUSSION TITLE
Sorting issues by creation date in GraphQL from GitHub does not work

### DISCUSSION THREAD

**Discussion by Vi-812** [community/community#49996] [Projects and Issues]
### Select Topic Area

Bug

### Body

```
query MyQuery {
  repository(name: "dbeaver", owner: "dbeaver") {
    issues(first: 100, filterBy: {labels: "bug", createdBy: "2022-01-01..*"}) {
      edges {
        node {
          createdAt
          labels(first: 100) {
            edges {
              node {
                name
              }
            }
          }
        }
      }
      totalCount
    }
  }
}
```
Sorting issues by creation date in GraphQL from GitHub does not work. It is very strange, no matter what date I enter, in the example above there will be 9 issues. If I enter "2022", it does not find any issue.
Please tell me what I'm doing wrong? Or is it really a bug. Thanks!

**Comment by Anmol-Baranwal** [comment]
It seems like you're not specifying a sort order for the issues in your query. By default, GitHub sorts issues by their relevance to the search query, rather than by creation date.

To sort the issues by creation date, you can add a sort parameter to your issues field, like this:

```php
query MyQuery {
  repository(name: "dbeaver", owner: "dbeaver") {
    issues(first: 100, filterBy: {labels: "bug", createdBy: "2022-01-01..*"}, orderBy: {field: CREATED_AT, direction: ASC}) {
      edges {
        node {
          createdAt
          labels(first: 100) {
            edges {
              node {
                name
              }
            }
          }
        }
      }
      totalCount
    }
  }
}
```
In this example, I've added an `orderBy` parameter to the issues field, specifying that the `issues` should be sorted by their `CREATED_AT` field in ascending order.

Note that GitHub's GraphQL API has a default maximum limit of 100 items per page, so if you have more than 100 issues that match your query, you may need to use pagination to retrieve all of them.

**Reply by Vi-812** [reply]
Unfortunately, your solution doesn't work. Your code also always gives 9 issues in response. Sorting does not change the output. [You can check the code here](https://docs.github.com/ru/graphql/overview/explorer).

**Reply by Vi-812** [reply]
I don't understand the question (You can generate a ticket right?). At the moment my code contains
`issues(first: 100, filterBy: {labels: $labels, since: $date}, after: $cursor) {`
and works perfectly. But `since` is not what I need. I need a `createdBy` that doesn't work correctly.

**Comment by Jeniferloras** [comment]
Here are a few steps that you can try to solve the issue:
1.	Check the GraphQL query that you are using to sort the issues. Make sure that you are using the correct syntax for sorting by creation date. The correct syntax is:

> php
> query {
>   repository(owner: "owner", name: "repositoryName") {
>     issues(first: 10, orderBy: {field: CREATED_AT, direction: ASC}) {
>       edges {
>         node {
>           title
>           createdAt
>         }
>       }
>     }
>   }
> }
> 

2.	If you are using the correct syntax and still facing issues with sorting by creation date, check whether you have the correct permissions to access the issues in the repository. You should have at least read permissions to access the issues.
3.	Make sure that you have the latest version of the GraphQL API for GitHub. You can check the current version of the API in the GitHub [DGME](https://dolgeneraldgme.com) documentation.
4.	If the issue persists, try using a different sorting method, such as sorting by title or by the number of comments, to check if the issue is specific to sorting by creation date.
5.	You can also try using a GraphQL client like GraphiQL or Altair to run the query and see if there are any errors in the response.
If none of the above solutions work, you can try reaching out to the GitHub support team for further assistance.

**Reply by Vi-812** [reply]
I don't understand what this is all about. I have a problem with filtering, not sorting. The request does not work correctly not only in my program, [but also in Explorer GitHub](https://docs.github.com/ru/graphql/overview/explorer). 
A tip from the "turn it off and on" series.

**Reply by Vi-812** [reply]
Without a paid account, I can't contact GitHub support directly, as far as I understand.

**Reply by Anmol-Baranwal** [reply]
You can generate a ticket right? I will look into my mistake and will tell you for sure.
